util/src/main/java/tech/pegasys/artemis/util/collections/LimitedMap.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.util.collections;
+
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class LimitedMap {
+  private LimitedMap() {}
+
+  /**
+   * Creates a limited map with a default initial capacity.
+   *
+   * @param maxSize The maximum number of elements to keep in the map.
+   * @param mode A mode that determines which element is evicted when the map exceeds its max size.
+   * @param <K> The key type of the map.
+   * @param <V> The value type of the map.
+   * @return A thread-safe map that will evict elements when the max size is exceeded.
+   */
+  public static final <K, V> Map<K, V> create(final int maxSize, final LimitStrategy mode) {
+    return create(16, maxSize, mode);
+  }
+
+  /**
+   * @param initialCapacity The initial size to allocate for the map.
+   * @param maxSize The maximum number of elements to keep in the map.
+   * @param mode A mode that determines which element is evicted when the map exceeds its max size.
+   * @param <K> The key type of the map.
+   * @param <V> The value type of the map.
+   * @return A thread-safe map that will evict elements when the max size is exceeded.
+   */
+  public static final <K, V> Map<K, V> create(
+      final int initialCapacity, final int maxSize, final LimitStrategy mode) {
+    final boolean useAccessOrder = mode.equals(LimitStrategy.DROP_LEAST_RECENTLY_ACCESSED);
+    return Collections.synchronizedMap(
```

Comment:
I'm not sure we should move the `synchronizedMap` into here.  I think it makes sense to potentially have instances of `LimitedMap` that are only accessed by a single thread or where there are already other locking mechanisms in place.  It also follows the typical pattern of collection classes not being thread-safe unless they have `Concurrent` in the name.

Reply:
I see that we'd previously had the `synchronizedSet` applied automatically to `LimitedSet` so if we did move this we'd have to ensure we added that to each place `LimitedSet` is constructed.